lib/bundler/compact_index_client/cache.rb

```diff
@@ -92,19 +94,9 @@ def lines(path)
         header ? lines[header + 1..-1] : lines
       end
 
-      def parse_gem(string)
-        version_and_platform, rest = string.split(" ", 2)
-        version, platform = version_and_platform.split("-", 2)
-        dependencies, requirements = rest.split("|", 2).map {|s| s.split(",") } if rest
-        dependencies = dependencies ? dependencies.map {|d| parse_dependency(d) } : []
-        requirements = requirements ? requirements.map {|r| parse_dependency(r) } : []
-        [version, platform, dependencies, requirements]
-      end
-
-      def parse_dependency(string)
-        dependency = string.split(":")
-        dependency[-1] = dependency[-1].split("&") if dependency.size > 1
-        dependency
+      def parse_gem(line)
+        @dependency_parser ||= GemParser.new
```

Comment:
Could this just be a class method to save from having instance var & instance object?

Reply:
Do you mean defining methods in `GemParser` as class methods instead of instance methods?

I don't think that it's a good design. Class method only class is one of global objects. Less global objects are better.

If we need to use class method only class, we should use module like `Bundler::SharedHelpers`.

BTW, do we need the instance saving optimization here? Normally, there is one `Bundler::CompactIndexClient::Cache` object in a process. So I don't think that we don't need it.

Reply:
I agree with @kou. @colby-swandale can you mark this as resolved if you're happy with @kou's reasoning?